src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState, FormEvent } from 'react';
+import * as postService from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { ErrorType } from '../../types/Error';
+import { FilterType } from '../../types/FilterType';
+import { Footer } from '../../components/Footer/Footer';
+import { TodoItem } from '../../components/TodoItem/TodoItem';
+import cs from 'classnames';
+import { USER_ID } from '../../api/todos';
+
+export const TodoList: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    postService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setError(ErrorType.LoadTodos);
+        setTimeout(() => setError(null), 3000);
+      });
+  }, []);
+
+  function handleSubmit(event: FormEvent) {
+    event.preventDefault();
+
+    const trimmedTitle = newTodoTitle.trim();
+    if (!trimmedTitle) {
+      setError(ErrorType.NoTitle);
+      setTimeout(() => setError(null), 3000);
+      return;
+    }
+
+    setIsAdding(true);
+    setTempTodo({
+      id: 0,
+      title: trimmedTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    postService.createTodo({ title: trimmedTitle, completed: false, userId: USER_ID })
+      .then((newTodo) => {
+        setTodos((currentTodos) => [...currentTodos, newTodo]);
+        setTempTodo(null);
+        setNewTodoTitle('');
+      })
+      .catch(() => {
+        setError(ErrorType.AddTodo);
+      })
+      .finally(() => {
+        setIsAdding(false);
+      });
+  }
+
+  const clearCompleted = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    Promise.all(completedTodos.map(todo => postService.deleteTodo(todo.id)))
+      .then(() => {
+        setTodos(todos.filter(todo => !todo.completed));
+      })
+      .catch(() => {
+        setError(ErrorType.DeleteTodo);
+        setTimeout(() => setError(null), 3000);
+      });
+  };
+
+  function deleteTodo(todoId: number) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const deleteTodo = (todoId: number) => {
```